### PR TITLE
[FLINK-24271][docs] Add document for special char in JSON_VALUE and JSON_QUERY

### PR DESCRIPTION
**Pull request by hililiwei**
## What is the purpose of the change

Add document for special char in JSON_VALUE and JSON_QUERY

https://issues.apache.org/jira/browse/FLINK-24271

## Brief change log

Add document for special char in JSON_VALUE and JSON_QUERY

## Verifying this change

document 

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/data/sql_functions.yml
Comment (Airblader) [inline docs/data/sql_functions.yml:670]: Since FLINK-24271 is about adding an example on how to access keys with "non-trivial" names, how about keeping the example as simple as possible to focus on actually showing how to do that? For example, like this:

```
// INT: 1
JSON_VALUE('{"a.b": 1}', '$.[''a.b'']' RETURNING INT)
```

Please also run the docs locally and make sure that the line length of the example doesn't cause the code block to scroll. If it does, line-break the example (as I did in the examples above) to prevent it from scrolling.

Comment (Airblader) [reply docs/data/sql_functions.yml:670]: For what it's worth, there is nothing special about this in terms of JSON functions. This is just a case of escaping a single quote within a SQL string.

File: docs/data/sql_functions.yml
Comment (Airblader) [inline docs/data/sql_functions.yml:693]: I think one example for `*` would be enough, and again it would be nice to make the example simple so it focuses on what we actually want to demonstrate.

Also, the comment showing the result should indicate the proper type, i.e. put quotes around it to signify that it returns a string and not an array:

```
-- '[1, 2, 3]'
JSON_QUERY('{"a": [1, 2, 3]}', 'lax $.a[*]')
```

File: docs/data/sql_functions.yml
Comment (Airblader) [inline docs/data/sql_functions.yml:669]: Could you maybe turn all of the `//` for the examples of this function into `--`? The double slashes are actually incorrect for SQL, so we should fix this in a separate commit.

**Review by Airblader — CHANGES_REQUESTED** [review]
Thanks for the pull request, @hililiwei! Can you please also update the JavaDocs, ScalaDocs and PyFlink docs so that all examples stay in sync?

Also, the code blocks in the examples can currently scroll if the line length of an example is too long. Since the UX for that is currently really bad (we have JIRA issue to improve the function docs) we need to manually make sure the examples have line breaks to not cause the code block to scroll horizontally.

File: docs/data/sql_functions.yml
Comment (hililiwei) [reply docs/data/sql_functions.yml]: done, only one is kept.

**Comment by hililiwei** [comment]
> Thanks for the pull request, @hililiwei! Can you please also update the JavaDocs, ScalaDocs and PyFlink docs so that all examples stay in sync?
> 
> Also, the code blocks in the examples can currently scroll if the line length of an example is too long. Since the UX for that is currently really bad (we have JIRA issue to improve the function docs) we need to manually make sure the examples have line breaks to not cause the code block to scroll horizontally.

hi @Airblader, I added example in JavaDocs & Pydocs, but where should I add the example of scaladocs?

File: flink-python/pyflink/table/expression.py
Comment (Airblader) [inline flink-python/pyflink/table/expression.py:1536]: This wouldn't compile.

```suggestion
            >>> lit('{"a.b": [0.998,0.996]}').json_value("$.['a.b'][0]", \
```

**Review by Airblader — CHANGES_REQUESTED** [review]
Thanks @hililiwei. You're right, there's nothing we need to do for ScalaDocs here. I left one last comment for the Python docs. After that I think this can be merged.

File: docs/data/sql_functions.yml
Comment (Airblader) [inline docs/data/sql_functions.yml:670]: I was actually just thinking: doesn't using double quotes work, too, and would be easier than having to escape the single quotes?

```
JSON_VALUE('{"a.b": [0.998,0.996]}','$.["a.b"][0]' 
```

Comment (hililiwei) [reply docs/data/sql_functions.yml:670]: Double quotation marks to get the correct result. If I was a developer unfamiliar with SQL, I might think double quotes are better understood.

Comment (Airblader) [reply docs/data/sql_functions.yml:670]: I think even SQL-savy people would expect this here, because this is a JSON function, and in JSON you use double quotes. So double quotes actually kind of make more sense to me than trying to use single quotes.

Comment (hililiwei) [reply docs/data/sql_functions.yml:670]: so, in these examples, should we use double quotes instead?

Comment (Airblader) [reply docs/data/sql_functions.yml:670]: I think it would be the better alternative to suggest, so let's change it, yes.

**Review by Airblader — APPROVED** [review]
LGTM now, thank you. Maybe you could squash the commits? We'll need a committer to get this merged then.

**Comment by hililiwei** [comment]
@Airblader In addition, should we create an issue (sub task of https://issues.apache.org/jira/browse/FLINK-9477 ）to translate "JSON Function" pages into Chinese?

**Comment by hililiwei** [comment]
Hi,@JingsongLi ,any other thoughts?can u help get this merged?

**Comment by Airblader** [comment]
@hililiwei Would you be able to do that translation in the end? I don't speak Chinese, so I wouldn't be able to do that task. It'd be good if we could close the parent issue ultimately, but I'm not 100% sure how Chinese translations should be handled here either. My understanding is that it's just an on-going effort.

**Comment by hililiwei** [comment]
> @hililiwei Would you be able to do that translation in the end? I don't speak Chinese, so I wouldn't be able to do that task. It'd be good if we could close the parent issue ultimately, but I'm not 100% sure how Chinese translations should be handled here either. My understanding is that it's just an on-going effort.

Let me open another issue to translate this part of the content. Although I use Chinese, my translation skills are not good enough. Let's see if there's someone else to do the translation.

**Review by JingsongLi — APPROVED** [review]
Looks good to me. Thanks @hililiwei and @Airblader

**Pull request merged**
